### ISSUE TITLE
Assertion violation at nla_basics_lemmas.cpp:262 (arith.solver=6)

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#3637]
Hi, for the following formula

[262.txt](https://github.com/Z3Prover/z3/files/4412887/262.txt)

z3 (commit 1d6fb6352fb5def) throws an assertion violation
~~~~
File: ../src/math/lp/nla_basics_lemmas.cpp
Line: 262
!c().check_monic(c().emons()[v])
~~~~

Options
~~~~
(set-logic QF_NRA)
(set-option :smt.arith.bprop_on_pivoted_rows false)
(set-option :smt.arith.solver 6)
(set-option :smt.phase_selection 5)
~~~~

**Comment by zhendongsu** [comment]
Another much smaller repro for the same crash (no options other than smt.arith.solver=6): 

```
[634] % z3 smt.arith.solver=6 small.smt2
ASSERTION VIOLATION
File: ../src/math/lp/nla_basics_lemmas.cpp
Line: 262
!c().check_monic(c().emons()[v])
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
a
[635] % 
[635] % cat small.smt2
(declare-fun a () Real)
(declare-fun b () Real)
(declare-fun c () Real)
(declare-fun d () Real)
(assert (<= 0.0 d (- b 1.0) (mod (to_int (* (- b c) a)) (to_int c))))
(check-sat)
[636] % 
```

**Comment by levnach** [comment]
Fixed in 0dc5bad6e4300113e1d3e7f393762cb24d72ad29
